### DISCUSSION TITLE
Building Dockerfiles

### DISCUSSION THREAD

**Discussion by dkoci** [quay/quay#1012] [Q&A] [answered]
Hi.
What is the difference between "building Dockerfiles with Build workers" and "Building Dockerfiles".
Is it possible to build Dockerfiles without kubernetes/openshif via gitlab webhook?

![image](https://user-images.githubusercontent.com/5565897/144292928-3af5f1c1-e612-4384-8d89-b9983f77d112.png

**Comment by flavianmissi** [comment, marked as answer]
Hi @dkoci,
There is no difference between the two. The documentation session "Automatically building Dockerfiles with Build workers" explains how the builders work, while the "Building Dockerfiles" session explain how to use the feature in quay.
The build manager currently needs to run on a bare-metal cluster. We're changing that on [3.7](https://issues.redhat.com/browse/PROJQUAY-295), keep an eye out for it!

**Reply by dkoci** [reply]
My openshift cluster (quay is running on openshift) is running on vmware. So I am not be able to use quay builder in current quay version?

**Reply by flavianmissi** [reply]
Yes, on 3.6.x and earlier you need to run the builder on a bare metal worker node, the docs for running builders on openshift[1] explains it in more detail.

[1] https://docs.projectquay.io/use_quay.html#openshift-requirements

**Reply by dkoci** [reply]
I am using openshift 4.9.

**Reply by flavianmissi** [reply]
Oh, I mean _quay_ 3.6.x and earlier. If you have the ability of deploying a bare metal worker node on your openshift cluster you will be able to use builders. Otherwise you need to wait until quay 3.7 is out.
